### PR TITLE
Simplify the default configuration file 

### PR DESCRIPTION
**Pull request by ashera96**
### Purpose
To limit the config entries maintained under `config.toml` file. 

This PR introduces a template file for the `config.toml`, `config.toml.template` in an effort to maintain limited number of configs under the default configuration file.

### Issues

Fixes [#4162](https://github.com/wso2-enterprise/choreo/issues/4162)

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/config/default_config.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+			MaximumSize: 10000,
+			ExpiryTime:  15,
+		},
+		Throttling: throttlingConfig{
+			EnableGlobalEventPublishing:        false,
+			EnableHeaderConditions:             false,
+			EnableQueryParamConditions:         false,
+			EnableJwtClaimConditions:           false,
+			JmsConnectionInitialContextFactory: "org.wso2.andes.jndi.PropertiesFileInitialContextFactory",
+			JmsConnectionProviderURL:           "amqp://admin:$env{tm_admin_pwd}@carbon/carbon?brokerlist='tcp://apim:5672'",
```
Comment (VirajSalaka) [inline adapter/config/default_config.go:169]: Are we going to stick to the `apim` host name or `localhost`

**Review by VirajSalaka — APPROVED** [review]

File: adapter/config/default_config.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chashikajw) [inline adapter/config/default_config.go:2]: Change this to 2021

File: resources/conf/config-for-eventhub.toml [hunk 3]
```diff
@@ -1,264 +1,92 @@
...
-    certificateAlias = "mgw"
-    # URL of the JWKs endpoint
-    jwksURL = ""
-    # Validate subscribed APIs
-    validateSubscription = false
-    # The claim in which the consumer key of the application is coming
-    consumerKeyClaim = ""
-    # Certificate Filepath within enforcer
-    certificateFilePath = "/home/wso2/security/truststore/mg.pem"
+  name="MGW"
```
Comment (chashikajw) [inline resources/conf/config-for-eventhub.toml:67]: Shall we add spaces here

File: resources/conf/config.toml [hunk 4]
```diff
@@ -1,265 +1,81 @@
...
-    certificateAlias = "mgw"
-    # URL of the JWKs endpoint
-    jwksURL = ""
-    # Validate subscribed APIs
-    validateSubscription = false
-    # The claim in which the consumer key of the application is coming
-    consumerKeyClaim = ""
-    # Certificate Filepath within enforcer
-    certificateFilePath = "/home/wso2/security/truststore/mg.pem"
+  name="MGW"
```
Comment (chashikajw) [inline resources/conf/config.toml:57]: same here

File: enforcer/src/main/java/org/wso2/choreo/connect/enforcer/config/ConfigHolder.java [hunk 5]
```diff
@@ -298,13 +294,6 @@ private void loadTrustStore() {
         }
     }
 
-    private void populateAPIMCredentials(AmCredentials cred) {
-        String username = cred.getUsername();
-        char[] password = cred.getPassword().toCharArray();
-        CredentialDto credentialDto = new CredentialDto(username, password);
```
[the remark this answers is not in the document]
Comment (ashera96) [reply enforcer/src/main/java/org/wso2/choreo/connect/enforcer/config/ConfigHolder.java:304]: We do have a usage

**Review by praminda — APPROVED** [review]

File: api/proto/wso2/discovery/config/enforcer/config.proto [hunk 6]
```diff
@@ -29,8 +28,6 @@ message Config {
 
     Service authService = 5;
 
-    AmCredentials apimCredentials = 6;
```
[the remark this answers is not in the document]
Comment (ashera96) [reply api/proto/wso2/discovery/config/enforcer/config.proto:32]: Pulled the other fields up.

[the remark this answers is not in the document]
Comment (ashera96) [reply api/proto/wso2/discovery/config/enforcer/config.proto:32]: Pulled the other fields up.

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**